spec-tests/src/test/java/org/candlepin/spec/CloudRegistrationSpecTest.java

```diff
@@ -447,6 +453,155 @@ public void shouldNotExportCertificatesWithAnonymousTokenForUnknownAnonCloudCons
+            .createConsumer(Consumers.random(ownerDTO)))
+            .isTooMany()
+            .hasHeaderWithValue("retry-after", 300);
+
+        adminClient.hosted().createOwner(ownerDTO);
+        associateOwnerToCloudAccount(adminClient, accountId, ownerDTO.getKey());
+
+        // The Org exists upstream of Candlepin, but does not have SKU(s) subscribed to it yet
+        assertThatStatus(() -> ApiClients.bearerToken(result.getToken()).consumers()
+            .createConsumer(Consumers.random(ownerDTO)))
+            .isTooMany()
+            .hasHeaderWithValue("retry-after", 180);
+
+        adminClient.hosted().createSubscription(Subscriptions.random(ownerDTO, productDTO));
+
+        // No product and no owner exist in Candlepin
+        assertThatStatus(() -> ApiClients.bearerToken(result.getToken()).consumers()
+            .createConsumer(Consumers.random(ownerDTO)))
+            .isTooMany()
+            .hasHeaderWithValue("retry-after", 60);
+
+
+        // Just owner exist in Candlepin
+        adminClient.owners().createOwner(ownerDTO);
+        assertThatStatus(() -> ApiClients.bearerToken(result.getToken()).consumers()
+            .createConsumer(Consumers.random(ownerDTO)))
+            .isTooMany()
+            .hasHeaderWithValue("retry-after", 30);
+
+        // Product and owner exist in Candlepin
+        adminClient.ownerProducts().createProductByOwner(ownerDTO.getKey(), productDTO);
+        adminClient.owners().createPool(ownerDTO.getKey(), Pools.random(productDTO));
+        ConsumerDTO consumer = ApiClients.bearerToken(result.getToken()).consumers()
+            .createConsumer(Consumers.random(ownerDTO));
+
+        assertThat(consumer)
+            .isNotNull()
+            .extracting(ConsumerDTO::getIdCert)
+            .isNotNull();
+    }
+
+    /**
+     * Associates a cloud offering ID to product IDs in the hosted test adapters.
+     *
+     * @param client
+     *     client used to make the request to the hosted test endpoint
+     *
+     * @param cloudOfferId
+     *     the offering ID to associate to a product ID
+     *
+     * @param productIds
+     *     the product IDs to associate to an offering ID
+     */
+    private void associateProductIdToCloudOffer(ApiClient client, String cloudOfferId,
+        Collection<String> productIds) {
+        ObjectNode objectNode = ApiClient.MAPPER.createObjectNode();
+        objectNode.put("cloudOfferId", cloudOfferId);
+
+        ArrayNode productsNode = ApiClient.MAPPER.createArrayNode();
+        productIds.forEach(prod -> productsNode.add(prod));
+        objectNode.putPOJO("productIds", productsNode);
+
+        Response response = Request.from(client)
+            .setPath("/hostedtest/cloud/offers")
+            .setMethod("POST")
+            .setBody(objectNode.toString())
+            .execute();
+
+        if (response.getCode() != 204) {
+            throw new RuntimeException("unable to associate product to cloud offer");
+        }
+    }
```

Comment:
This was moved into the `HostedTestAPI` so you can just use `adminClient.hosted().associateProductIdsToCloudOffer` instead

Reply:
done

Reply:
It looks like you are still using this method in a couple tests. Once you are using just `adminClient.hosted().associateProductIdsToCloudOffer` you should be able to remove this method

Reply:
oh sorry i missed that they are two diffrents endpoints. Should be done